### PR TITLE
Some more! ^^

### PR DESCRIPTION
**Pull request by Igigog**
This one is quite a lot. Since I'm finished with what I wanted to achieve, feel free to take your time reviewing the changes.
It builds on top of the other pull request, so better review it first.

Changelog:
1. Major rust code refactoring
2. I moved all commands into its own interface. `send_cmd` is now type-safe and won't send just random byte arrays, unless you really try.
3. I let Tauri take care of serialization and deserialization instead of returning strings. Works like a charm.
4. Added some very basic tests for commands. Since I don't actually know how they work, you may want to write something better, but these ones will do for now.
5. Fixed all `cargo clippy` warnings. It took some minor renaming, so I also added the config migration. Don't know if it actually works fully, but my config was fine.
6. Added error reporting to frontend. Now most errors will actually be reported.
7. Added a very simple config validation. It will not let you set `quality` of a filter to 0 and destroy your ears anymore. Although it is _very_ simple, together with error reporting it works surprisingly user-friendly.
8. Made the window resizable.
9. Condensed the frontend a bit. Now a full screen Toolbox shows a whole 3 filters, which isn't much but kinda better than 2. Also, looks fancy.

It's not perfect of course, but it sure got better to work with. At least, now Toolbox won't actively try to send a bad config right into your ears ^^. And my work is done here, I've implemented all I've wanted so far.

### CODE REVIEW

**Comment by Igigog** [comment]
oops, wrong branch actually

**Comment by Igigog** [comment]
now it's right

File: src-tauri/src/main.rs [hunk 1]
```diff
@@ -94,582 +105,483 @@ fn find_configuration_endpoints<T: UsbContext>(
...
-#[derive(Serialize, Deserialize, Default)]
-struct Preprocessing {
-    preamp: f32,
-    postEQGain: f32,
-    reverse_stereo: bool
+    fn to_payload(&self) -> Vec<u8> {
+        let mut preprocessing_payload: Vec<u8> = Vec::new();
+        // TODO: -1.0 as the firmware adds 1, cleanup later. Consider storing this value without the subtraction
+        // to eliminate a math op and make the code more grokable?
+        //preprocessing_payload.extend_from_slice(&(f32::powf(10.0, cfg.preprocessing.preamp/20.0) - 1.0).to_le_bytes());
```
Comment (george-norton) [inline src-tauri/src/main.rs:156]: I guess we can delete this commented out line?

Comment (Igigog) [reply src-tauri/src/main.rs:156]: Yeah, looks not important, let's delete it

**Pull request merged**